include/boost/date_time/time_resolution_traits.hpp

```diff
@@ -68,7 +68,7 @@ namespace date_time {
            typename frac_sec_type::int_type resolution_adjust,
 #endif
            unsigned short frac_digits,
-           typename var_type = std::time_t >
```

Comment:
The type var_type defaults to a value, but that can be overridden. Does this affect the serialization code which you have corrected ? In other words can the end-user change the time_duration internal types by configuration time_resolution_traits differently ? If so, in that case isn't the correct code always to use the time_resolution_traits var_type as the type used to serialize rather than doing asserts to make sure the type is always the var_type default ?

Reply:
So if we had the static assertions in the code before this, and they all used var_type, they would all have passed when var_type changed.  This would have defeated the purpose of the static asserts, which is to protect the binary format compatibility.  In Boost 1.65.1 and earlier var_type was `boost::int32_t` which had the year 2038 problem.  In 1.66.0 the var_type changed to `std::time_t`, thus causing it to vary on different platforms.  By changing it to boost::int64_t and ensuring the output size is 64-bits with static assertions, it protects the binary serialization.

Reply:
You should put that comment "_In Boost 1.65.1 and earlier var_type was boost::int32_t which had the year 2038 problem. In 1.66.0 the var_type changed to std::time_t, thus causing it to vary on different platforms. By changing it to boost::int64_t and ensuring the output size is 64-bits with static assertions, it protects the binary serialization._" in the source code, so that people will know why it is this way.

Reply:
Okay, will do and push later.

Reply:
Can the var_type be changed by the end-user ? Or is time_resolution_traits strictly used internally ? if the latter I have no objection to the serialization code in this fix. But if time_resolution_traits, and therefore var_type, can be used by the end-user to change the time duration types then I think the serialization code must use the var_type rather than assume a hardcoded setting which is merely the default type for the var_type.

Reply:
I believe posix_time_config.hpp is the only consumer, and it relies on the default value.